Cyber-Mint/wg-vpn
# Maintenance and minor improvements

### Description
General improvements, upgrades, and maintenance of deployment scripts and backend.

---
### Changes
- Used ansible jinja templating for wg0.conf and nginx.conf instead of copying over files and replacing variables with regex expressions
- Upgraded Python dependencies
- Separate linting jobs in circleci config to run as separate jobs
- Added new 404 error page (previously responded with JSON response)
- Fixed minor typos in deploy and comments
- Removed CertBot code from nginx.conf (this code will be added by cerbot during the letsencrypt role)

---
### Notes on backward compatibility
Fully backward compatible. The code changed in the backend is tested by postman to verify no regression with the version upgrades. These version upgrades are to the docker container, and hence are safely portable for replacing existing deployments. All other changes to static pages and deployment are improvements and automating upgrades will be performed on redeploy. 

---
### How To Test
**BE**: Ensure that the pipeline passes
**Ansible scripts**: Deploy ansible scripts to a VM or instance. These changes will be "loud" if they break (i.e. the deployment will not be successful)

> **Notes**: It is not necessary to test the full functionality, the nature of the changes allow afford us to check only that:
> - The deployment passes without error
> -  No regression on the BE APIs
